src/getting-started.md

```diff
@@ -0,0 +1,332 @@
+
+[For more details on this process, see this chapter.](./implementing_new_features.md)
+
+### Breaking Changes
+
+As mentioned above, Rust has strong backwards-compatibility guarantees. To this
+end, we are reluctant to make breaking changes. However, sometimes they are
+needed to correct compiler bugs (e.g. code that compiled but should not).
+
+Depending on the scale of the breakage, there are a few different actions that
+can be taken.  If the reviewer believes the breakage is very minimal (i.e. very
+unlikely to be actually encountered by users), they may just merge the change.
+More often, they will request a Final Comment Period (FCP), which calls for
+rough consensus among the members of a relevant team. The team members can
+discuss the issue and either accept, reject, or request changes on the PR.
+
+If the scale of breakage is large, a deprecation warning may be needed. This is
+a warning that the compiler will display to users whose code will break in the
+future.  After some time, an FCP can be used to move forward with the actual
+breakage.
+
+If the scale of breakage is unknown, a team member or contributor may request a
+[crater] run. This is a bot that will compile all crates.io crates and many
+public github repos with the compiler with your changes. A report will then be
+generated with crates that ceased to compile with or began to compile with your
+changes. Crater runs can take a few days to complete.
+
+[crater]: https://github.com/rust-lang/crater
+
+### Major Changes
+
+The compiler team has a special process for large changes, whether or not they
+cause breakage. This process is call Major Change Proposal (MCP). MCP is a
+relatively lightweight mechanism for getting feedback on large changes to the
+compiler (as opposed to a full RFC or a design meeting with the team).
+
+Example of things that might require MCPs include major refactorings, changes
+to important types, or important changes to how the compiler does something.
+
+**When in doubt, ask on [zulip][z]. We would hate for you to put a lot of work
+into a PR that ends up not getting merged!**
+
+### Performance
+
+Compiler performance is important. We have put a lot of effort over the last
+few years into [gradually improving it][perfdash].
+
+[perfdash]: https://perf.rust-lang.org/dashboard.html
+
+If you suspect that your change may cause a performance regression (or
+improvement), you can request a "perf run" (your reviewer may also request one
+before approving). This is yet another bot that will compile a collection of
+benchmarks on a compiler with your changes. The numbers are reported
+[here][perf], and you can see a comparison of your changes against the latest
+master.
+
+[perf]: https://perf.rust-lang.org/dashboard.html
+
+## Other Resources
+
+- This guide: talks about how `rustc` works
+- [The t-compiler zulip][z]
+- [The compiler's documentation (rustdocs)](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/)
```

Comment:
```suggestion
- [The compiler's documentation (rustdocs)](https://doc.rust-lang.org/nightly/nightly-rustc/)
```

Reply:
Hm, I think the current is fine.

Reply:
But `rustc_middle` is just one part of the compiler; it doesn't include parsing, AST, etc.

Reply:
It's just an entry point, we can search everything in the compiler from there. But yeah I didn't know the current nightly-rustc doc has the index page. If we accept this, we could also update the same links in this guide to simplify.

Reply:
That's true. I honestly have no idea why, but we always link to `rustc_middle`. I am willing to change this, but I am curious to hear what the others say.

Reply:
Huh, I didn't realize that! Perhaps the reason why is as the crate describes it:

> The "main crate" of the Rust compiler. This crate contains common type definitions that are used by the other crates in the rustc "family".

Reply:
> I honestly have no idea why, but we always link to rustc_middle

IIRC we didn't have the index page until recently. And it was `/nightly-rustc/rustc` before renaming so people may have thought _it was the index page_.

Reply:
Which crate was `/nightly-rustc/rustc`? `rustc_middle`?

Reply:
> Which crate was /nightly-rustc/rustc? rustc_middle?

We renamed `librustc` to `librustc_middle` a while ago.